lib/mutant/ast/regexp/transformer/quantifier.rb

```diff
@@ -0,0 +1,97 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        class Quantifier < self
+          %i[
+            regexp_greedy_zero_or_more
+            regexp_greedy_one_or_more
+            regexp_greedy_zero_or_one
+            regexp_possessive_zero_or_one
+            regexp_reluctant_zero_or_more
+            regexp_reluctant_one_or_more
+            regexp_possessive_zero_or_more
+            regexp_possessive_one_or_more
+            regexp_greedy_interval
+            regexp_reluctant_interval
+            regexp_possessive_interval
+          ].map(&method(:register))
+
+          class ExpressionToAST < Transformer::ExpressionToAST
+            def call
+              ast(expression.min, expression.max)
+            end
+
+          private
+
+            def ast_type
+              :"regexp_#{expression.mode}_#{expression.token}"
+            end
+          end
+
+          class ASTToExpression < Transformer::ASTToExpression
+            include NamedChildren
+
+            children :min, :max, :subject
+
+            QUANTIFIER_MAP = IceNine.deep_freeze(
+              regexp_greedy_zero_or_more:     [:zero_or_more, '*',  :greedy],
```

Comment:
Instead of mapping to an array of 3 elements (effectively a tulpe) and having to remember the indexes each tuple member is represended with, what about using a dedicated value object?

```
class Quantifier
  include Concord::Public.new(:type, :suffix, :mode)
end
```ruby
QUANTIFIER_MAP = IceNine.deep_freeze({
  regexp_greedy_zero_or_more:     [:zero_or_more, '*',  :greedy]
}.map { |ast_type, arguments| [ast_type, Quantifier.new(*arguments)] }.to_h)
```

This will also remove or eliminate the "ugly" (due `Array#{first,last} non strictness) API usage in `#type`, `#mode`, `#suffix` implementations above.